framework/src/schema/application_config_schema.ts

```diff
@@ -12,10 +12,94 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+
+const moduleConfigSchema = {
+	type: 'object',
+	propertyNames: {
+		pattern: '^[a-zA-Z][a-zA-Z0-9_]*$',
+	},
+	additionalProperties: { type: 'object' },
+};
+
+const rewardsConfigSchema = {
+	type: 'object',
+	required: ['milestones', 'offset', 'distance'],
+	description: 'Object representing LSK rewards milestone',
+	properties: {
+		milestones: {
+			type: 'array',
+			items: {
+				type: 'string',
+				format: 'uint64',
+			},
+			description: 'Initial 5, and decreasing until 1',
+		},
+		offset: {
+			type: 'integer',
+			minimum: 1,
+			description: 'Start rewards at block (n)',
+		},
+		distance: {
+			type: 'integer',
+			minimum: 1,
+			description: 'Distance between each milestone',
+		},
+	},
+	additionalProperties: false,
+};
+
+const delegatesConfigSchema = {
+	type: 'array',
+	items: {
+		required: ['encryptedPassphrase', 'address', 'hashOnion'],
+		properties: {
+			encryptedPassphrase: {
+				type: 'string',
+				format: 'encryptedPassphrase',
+			},
+			address: {
+				type: 'string',
+				format: 'hex',
+			},
+			hashOnion: {
```

Comment:
doesn't this go to random module specific config?

Reply:
Yes I believe module specific config must be splitted. But I am not sure which one are those and when to do it.   @shuse2 may explain here.